### ISSUE TITLE
Crash initializing FileRef from stream

### ISSUE THREAD

**Issue by fillobotto** [taglib/taglib#836]
I am reading a file throug its stream I got from opening the file descriptor:
```
TagLib::IOStream * stream = new TagLib::FileStream((int) fd);
TagLib::FileRef f(stream);
```
Let alone the FileStream constructor I added for convenience. On the second line I got some report from users which devices crashed (Android environment). I don't have full information but I can somewhat "build" the backtrace:

> signal 11 (SIGSEGV), code 1 (SEGV_MAPERR) occurred [HERE](https://github.com/taglib/taglib/blob/taglib2/taglib/mpeg/mpegutils.h#L52)
>MPEG::isFrameSync(const ByteVector &bytes, unsigned int offset = 0)
> MPEG::File::isSupported(IOStream *stream)

I suppose that the cause is that it's not checking the length of ByteVector before accessing it.
Is it worth to add it? Does it mean the file is sort of corrupted?

**Comment by fillobotto** [comment]
Could it be related to PR #830?

**Comment by fillobotto** [comment]
I added a check inside the function and I can longer see those crashes:

```
      inline bool isFrameSync(const ByteVector &bytes, unsigned int offset = 0)
      {
        // 0xFF in the second byte is possible in theory, but it's very unlikely.
        if(bytes.isEmpty() || bytes.size() <= offset + 1) return false;
        const unsigned char b1 = bytes[offset + 0];
        const unsigned char b2 = bytes[offset + 1];
        return (b1 == 0xFF && b2 != 0xFF && (b2 & 0xE0) == 0xE0);
      }
```

**Comment by scotchi** [comment]
Yep, this is fixed by #830 as that adds a check for empty earlier in the body.
